Game Dev/Assets/Scripts/Player/WeaponSwitch.cs

```diff
@@ -0,0 +1,39 @@
+using UnityEngine;
+
+public class WeaponSwitch : MonoBehaviour
+{
+    public int SelectedWeapon;
+    
+    void Start()
+    {
+        SelectedWeapon = 0;
+        SelectWeapon();
+    }
+
+    void Update()
+    {
+        if (Input.GetAxis("Mouse ScrollWheel") > 0f)
+            SelectedWeapon = (SelectedWeapon + 1) % transform.childCount;
+        if (Input.GetAxis("Mouse ScrollWheel") < 0f)
+            SelectedWeapon = (SelectedWeapon + transform.childCount - 1) % transform.childCount;
+        SelectWeapon();
```

Comment:
No need to call switch at each frame, would be better if only called when the SelectedWeapon is changed.

Reply:
There is no need to be this precise, SelectWeapon() only iterates through an array of 3 elements so the time complexity is negligible even if we call the function on each frame or on each Mouse ScrollWheel.